serialization.md

```diff
@@ -75,7 +75,7 @@ This specification uses the following terms:
         A collection of tags grouped under a common prefix (the name component before <code>:</code>).
 	For example, in an image tagged with the name <code>my-app:3.1.4</code>, <code>my-app</code> is the <i>Repository</i> component of the name.
 	A repository name is made up of slash-separated name components, optionally prefixed by a DNS hostname.
-	The hostname must follow comply with standard DNS rules, but may not contain <code>_</code> characters.
+	The hostname must comply with standard DNS rules that apply to LDH or IDNA labels as defined in <a href="https://tools.ietf.org/html/rfc5890#section-2.3">RFC 5890</a>.
```

Comment:
From RFC 5890, [LDH labels include “Fake A-labels”][1], which I don't think we want to allow (at least, not if we are putting DNS-related constraints on hostnames, which is what this PR is about).  [Later in the RFC][2]:

> For IDNA-aware applications, the three types of valid labels are "A-labels", "U-labels", and "NR-LDH labels", each of which is defined below.

So I think we want to echo that and say:

> The hostname must be an A-label, U-label, or NR-LDH label, as defined in <a href="https://tools.ietf.org/html/rfc5890#section-2.3.2">RFC 5890</a>.

And we are not supporting [binary labels][3], which RFC 5890 lists as a [non-ASCII label not covered under IDNA][4]?  I'm personally agnostic on refusing or allowing binary labels, but want to make sure that we're making an explicit decision one way or the other.

[1]: https://tools.ietf.org/html/rfc5890#page-10
[2]: https://tools.ietf.org/html/rfc5890#section-2.3.2.1
[3]: https://tools.ietf.org/html/rfc2673
[4]: https://tools.ietf.org/html/rfc5890#page-11